packages/components/src/RadioItem.tsx

```diff
@@ -32,6 +32,9 @@ type RadioItemProps = {
 
   /** The value to associate with this radio item */
   value: string;
+
+  /** An id used for test */
+  dataTestid?: string;
```

Comment:
I think this is a case where we want to just allow the `data-testid` to be passed just like that:
```suggestion
  'data-testid'?: string;
```
I'm suggesting that because I was looking at other common attributes, such as the `aria-label`, and those are defined in the types using the kebab-case names: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b92cbb35c8cb01cc2150736ec8b8640c6af3b085/types/react/index.d.ts#L1558

It is an exception to how props are normally defined in JSX, but given the number of aria-* props and data-* props, I think we should stick with just naming it `data-testid`. 

We should probably define a type `type DataAttributes = { 'data-testid'?: string }` and then use that in pretty much all of our components (in addition to extending `InputHTMLAttributes<HTMLInputElement>` in this case).

I'm going to ask the team as well to ensure we're all on the same page for this.